### PR TITLE
Add water cups to water coolers

### PR DESCRIPTION
**Pull request by tgrkzus**
## About the PR

Added water cup dispensing to water coolers

As well as added examine on Bins (e.g. a paper bin) to say how many items are left.
And added an alt interact on bin interaction as a fallback to restocking items in bins.
Also to simplify things I make Water cups now default to empty, which is more sensible to me, why would a station have random water cups sitting around for hours? Minor nerf I guess to some maps?

Another note is you can technically put other cups in the bin, if you want. Which is a bit odd, but better than adding a random tag for just a paper cup I think.

## Why / Balance

Water coolers without cups are kinda useless in a lot of maps, other than as a random water tank.
The sprite has a paper cups on the side so I guess this was intended at some point.

## Technical details

- Reuses the same bin component as Paper Bin
- Use alt verbs to construct a fallback on bin interaction for restocking

## Media

Water cooler examine
watercooler

And demo of the usage.
https://github.com/space-wizards/space-station-14/assets/7283010/4232839d-daf8-4b0b-a3f2-b9e55ec20882

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Paper cups are now available at water coolers.

### CODE REVIEW

File: Resources/Locale/en-US/prototypes/entities/structures/storage/tanks/tanks.ftl [hunk 1]
```diff
@@ -14,7 +14,7 @@ ent-WaterTankFull = { ent-WaterTank }
     .desc = { ent-WaterTank.desc }
 
 ent-WaterCooler = water cooler
-    .desc = Seems like a good place to stand and waste time.
+    .desc = Seems like a good place to stand and waste time. It has a stock of paper cups on the side
```
Comment (Partmedia) [inline Resources/Locale/en-US/prototypes/entities/structures/storage/tanks/tanks.ftl:17]: Missing trailing period

**Comment by Partmedia** [comment]
Thank you for your contribution. I think this change is very welcome. I'm going to give this an in-game test in a bit and double check things work.

Please take a look at the test failure (which doesn't look like a false positive). If you have questions about what it means, please let us know.

File: Content.Shared/Storage/EntitySystems/BinSystem.cs [hunk 2]
```diff
@@ -31,6 +34,13 @@ public override void Initialize()
         SubscribeLocalEvent<BinComponent, EntRemovedFromContainerMessage>(OnEntRemoved);
         SubscribeLocalEvent<BinComponent, InteractHandEvent>(OnInteractHand);
         SubscribeLocalEvent<BinComponent, AfterInteractUsingEvent>(OnAfterInteractUsing);
+        SubscribeLocalEvent<BinComponent, GetVerbsEvent<AlternativeVerb>>(OnAltInteractHand);
+        SubscribeLocalEvent<BinComponent, ExaminedEvent>(OnExamined);
+    }
+
+    private void OnExamined(EntityUid uid, BinComponent component, ExaminedEvent args)
+    {
+        args.PushText(Loc.GetString("bin-component-on-examine-text", ("count", component.Items.Count)));
```
Comment (EmoGarbage404) [inline Content.Shared/Storage/EntitySystems/BinSystem.cs:43]: this locale string isn't in the files you've committed. did you forgor?

File: Content.Shared/Storage/EntitySystems/BinSystem.cs [hunk 3]
```diff
@@ -76,6 +86,20 @@ private void OnInteractHand(EntityUid uid, BinComponent component, InteractHandE
...
+    /// Alt interact acts the same as interacting with your hands normally, but allows fallback interaction if the item
+    /// has priority. E.g. a water cup on a water cooler fills itself on a normal click,
+    /// but you can use alternative interactions to restock the cup bin
+    /// </summary>
+    private void OnAltInteractHand(EntityUid uid, BinComponent component, GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (args.Using != null)
+        {
+            var canReach = args.CanAccess && args.CanInteract;
+            OnAfterInteractUsing(uid, component, new AfterInteractUsingEvent(args.User, (EntityUid) args.Using, args.Target, EntityCoordinates.Invalid, canReach));
```
Comment (EmoGarbage404) [inline Content.Shared/Storage/EntitySystems/BinSystem.cs:99]: you should really just make a private function with this code and call it in both. constructing a new event here is really strange.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

**Pull request merged**